From c468f6bec70b1b749e57c8e6ec09b6fe62d7cd6e Mon Sep 17 00:00:00 2001
From: Antoine Belvire <antoine.belvire@opensuse.org>
Date: Mon, 15 Aug 2022 23:31:00 +0200
Subject: [PATCH] Allow GUI to compile with wxWidgets built with STL

wxMenuItemList - a wxList of wxMenuItems - should be iterated using the
wxMenuItemList::compatibility_iterator so that it can be compiled with wxWidgets
built with STL.

This commit does not make GUI require wxWidgets > 2.8 nor wxWidgets compiled
with STL but makes the code compatible with more recent versions of wxWidgets
compiled with STL.

See also:
https://wiki.wxwidgets.org/Updating_to_the_Latest_Version_of_wxWidgets#wxUSE_STL.
---
 src/GUI/OBGUI.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/GUI/OBGUI.cpp b/src/GUI/OBGUI.cpp
index 2aa14dcb09..f65fe8cc1d 100644
--- a/src/GUI/OBGUI.cpp
+++ b/src/GUI/OBGUI.cpp
@@ -1128,7 +1128,7 @@ void OBGUIFrame::OnClickPlugin(wxCommandEvent& event)
   if(item)
   {
     //Find the name of the plugin type. It seems difficult to go up a menu hierarchy.
-    wxwxMenuItemListNode *node = listMenu->GetMenuItems().GetFirst();
+    wxMenuItemList::compatibility_iterator node = listMenu->GetMenuItems().GetFirst();
     while (node)
     {
       wxMenuItem* itemm = node->GetData();
